catroid/src/main/java/org/catrobat/catroid/content/actions/HideTextAction.java

```diff
@@ -37,20 +37,22 @@
 public class HideTextAction extends TemporalAction {
 
 	private UserVariable variableToHide;
+
+	private Scene scene;
+	private Sprite sprite;
 	private UserBrick userBrick;
 
 	@Override
 	protected void begin() {
-		DataContainer dataContainer = ProjectManager.getInstance().getCurrentScene().getDataContainer();
-		Sprite currentSprite = ProjectManager.getInstance().getCurrentSprite();
+		DataContainer dataContainer = scene.getDataContainer();
```

Comment:
You can use getSceneToPlay() here! With that you do not need the getSceneOfSprite Method.